pkg/hostagent/setup.go

```diff
@@ -281,23 +289,26 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 		} else {
 			mtu = agent.config.InterfaceMtu
 		}
-
+		//question: when can an IP be assigned before this? is it during a reallocation?
 		if len(iface.IPs) == 0 {
 			// We're doing ip address management
 
-			logger.Debug("Allocating IP address(es) for ", iface.Name)
+			//change back to debug
+			logger.Infof("Allocating IP address(es) for %v", iface.Name)
 			err = agent.allocateIps(iface, podKey)
 			if err != nil {
 				return nil, err
 			}
 		}
+		//Can we check if pod still exists here, and gracefully fail?
 		for _, ip := range iface.IPs {
 			//There are 4 cases: IPv4-only, IPv6-only, dual stack with either IPv4 or IPv6 as the first address.
 			//We are guaranteed to derive the MAC address from IPv4 if it is assigned
 			if ip.Address.IP != nil && ip.Address.IP.To4() != nil {
 				iface.HostVethName, iface.Mac, err =
 					runSetupVeth(iface.Sandbox, iface.Name, mtu, ip.Address.IP)
 				if err != nil {
+					deallocIP(iface)
```

Comment:
I think it will help to log here at info level that "runSetupVeth" has returned an (we can log the error as well) and that we are returning an error.

Reply:
We already log the error from runSetupVeth in the calling function https://github.com/noironetworks/aci-containers/blob/87ece9a8e4598995204b0041cbb7fe3de1df561e/pkg/hostagent/eprpc.go#L89.

Should we add another?